test/cli/test_login.rb

```diff
@@ -0,0 +1,272 @@
+    OUTPUT
+
+    # create login:
+    run_command "kbsecret new login -x #{label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # retrieve login:
+    run_command "kbsecret login #{label}" do |cmd|
+      cmd.wait
+      assert_equal output, cmd.output
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_terse
+    label = "test-terse"
+    username = "user"
+    password = "pass"
+    output = "#{label}:#{username}:#{password}"
+
+    # create login:
+    run_command "kbsecret new login -x #{label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # retrieve login tersely:
+    run_command "kbsecret login -x #{label}" do |cmd|
+      cmd.wait
+      assert_equal output, cmd.output.chomp
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_terse_separator
+    label = "test-terse-separator"
+    username = "user"
+    password = "pass"
+    separator = "^"
+    output = [label, username, password].join separator
+
+    # create login:
+    run_command "kbsecret new login -x #{label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # retrieve login with separator:
+    run_command "kbsecret login -i #{separator} -x #{label}" do |cmd|
+      cmd.wait
+      assert_equal output, cmd.output.chomp
+    end
+  ensure
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_all
+    session = "test-session"
+    label = "test-session-login"
+    label1 = "test-session-login1"
+    username = "user"
+    password = "pass"
+   # NOTE: need regex since we can't be sure of the order in which the logins will be retreived
+    pattern = /Label: (#{label}|#{label1})\n\tUsername: #{username}\n\tPassword: #{password}\nLabel: (#{label1}|#{label})\n\tUsername: #{username}\n\tPassword: #{password}\n/
+    error_pattern = /No such record/
+
+    # create a new session
+    run_command_and_stop "kbsecret session new -r test #{session}"
+
+    # fail to retrieve nonexistent logins:
+    run_command "kbsecret login -a" do |cmd|
+      cmd.wait
+      assert_match error_pattern, cmd.stderr
+    end
+
+    # create login:
+    run_command "kbsecret new login -s #{session} -x #{label}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # create another login:
+    run_command "kbsecret new login -s #{session} -x #{label1}" do |cmd|
+      cmd.stdin.puts "#{username}:#{password}"
+      cmd.stdin.close
+      cmd.wait
+    end
+
+    # retrieve logins:
+    run_command "kbsecret login -s #{session} -a" do |cmd|
+      cmd.wait
+      assert_match pattern, cmd.output
+    end
+  ensure
+    # remove session:
+    run_command_and_stop "kbsecret session rm -d #{session}"
```

Comment:
Make sure you delete the login records here, as well.

Reply:
unless something has changed, i believe that when you remove the session with -d, all of its records get destroyed

Reply:
Yep, you're right. Wasn't thinking 😄